skimage/segmentation/_slic.pyx

```diff
@@ -129,16 +140,22 @@ def _slic_cython(double[:, :, :, ::1] image_zyx,
                         dy = sy * (cy - y)
                         dy *= dy
                         for x in range(x_min, x_max):
+
+                            if mask_size > 0 and mask[z, y, x] == 0:
```

Comment:
I'm pretty sure C will be able to compile out this condition, but maybe not for all compile options.

Reply:
I agree. I added some brackets.